### PR TITLE
Introduce value test for quantized model

### PR DESCRIPTION
**Pull request by jinevening**
This introduces quantization-value-test for testing values of a quantized model

ONE-DCO-1.0-Signed-off-by: Hyukjin Jeong <hj1.jeong@samsung.com>

---
Related to #2088

### CODE REVIEW

File: compiler/quantization-value-test/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+
+add_custom_command(
+  OUTPUT ${TEST_CONFIG}
+  COMMAND ${CMAKE_COMMAND} -E remove -f ${TEST_CONFIG}
+  COMMAND ${CMAKE_COMMAND} -E echo 'RECORD_MINMAX_PATH=\"$<TARGET_FILE:record-minmax>\"' >> ${TEST_CONFIG}
+  COMMAND ${CMAKE_COMMAND} -E echo 'CIRCLE2CIRCLE_PATH=\"$<TARGET_FILE:circle2circle>\"' >> ${TEST_CONFIG}
+  COMMAND ${CMAKE_COMMAND} -E echo 'CIRCLE_TENSORDUMP_PATH=\"$<TARGET_FILE:circle-tensordump>\"' >> ${TEST_CONFIG}
+  DEPENDS record-minmax
+  DEPENDS circle2circle
+  COMMENT "Generate test configuration"
```
Comment (seanshpark) [inline compiler/quantization-value-test/CMakeLists.txt:101]: maybe add `DEPENDS circle-tensordump` ?

File: compiler/quantization-value-test/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+  # Copy .recipe
+  add_custom_command(OUTPUT "${RECIPE_SOURCE_FILE}"
+                     COMMAND ${CMAKE_COMMAND} -E copy_if_different
+                             "${TEST_REPO}/${RECIPE}" "${RECIPE_SOURCE_FILE}"
+                     DEPENDS "${TEST_REPO}/${RECIPE}"
+                     COMMENT "Generating ${RECIPE_SOURCE_FILE}")
+
+  # Generate .tflite
+  add_custom_command(OUTPUT "${RECIPE_OUTPUT_FILE}"
+                     COMMAND tflchef-file "${RECIPE_SOURCE_FILE}" "${RECIPE_OUTPUT_FILE}"
```
Comment (seanshpark) [inline compiler/quantization-value-test/CMakeLists.txt:75]: is it OK not using lie `$<TARGET_FILE:tflchef-file>` ?

File: compiler/quantization-value-test/CMakeLists.txt (hunk 1 above)
Comment (jinevening) [reply compiler/quantization-value-test/CMakeLists.txt:101]: My mistake. Thanks!

File: compiler/quantization-value-test/CMakeLists.txt (hunk 2 above)
Comment (jinevening) [reply compiler/quantization-value-test/CMakeLists.txt:75]: Yes, it is ok to use `tflchef-file`. Cmake can find `tflchef-file` during build stage.

File: compiler/quantization-value-test/README.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
+# quantization-value-test
+
+This test checks whether a Circle model (listed in `test.lst`) is correctly quantized. The circle models are generated from recipes saved in `res/TensorFlowLiteRecipes`.
+
+Write `test.local.lst` for local test.
+
+Test process
+Step 1. Fake quantization
```
Comment (mhs4670go) [inline compiler/quantization-value-test/README.md:8]: `\n` at the end of this line will be removed. Maybe that's not what you want I think. You should insert one more enter. You can check this `View file` in github or `Ctrl+Shft+v` in vscode

Comment (jinevening) [reply compiler/quantization-value-test/README.md:9]: Thank you for the tips. I've updated the README.

File: compiler/quantization-value-test/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+add_test(
+  NAME quantization_value_test
+  COMMAND "${CMAKE_CURRENT_SOURCE_DIR}/test_quantization.sh"
+          "${TEST_CONFIG}"
+          "${CMAKE_CURRENT_BINARY_DIR}"
+          ${QUANTIZATION_VALUE_TEST_WITH_PARAM}
+)
+
+set_tests_properties(record_minmax_value_test PROPERTIES DEPENDS fake_quantization_value_test)
+set_tests_properties(quantization_value_test PROPERTIES DEPENDS record_minmax_value_test)
```
Comment (parjong) [inline compiler/quantization-value-test/CMakeLists.txt:136]: Code itself looks good to me... but the name seems to be too general ``). Unfortunately, I can't suggest any better name T_T

File: compiler/quantization-value-test/README.md (hunk 3 above)
Comment (seanshpark) [reply compiler/quantization-value-test/README.md:9]: Maybe use like
```
### Test process

#### Step 1. Fake quantization
```
for better readability

Comment (jinevening) [reply compiler/quantization-value-test/README.md:9]: @seanshpark Looks much better. Thanks!

File: compiler/quantization-value-test/CMakeLists.txt (hunk 4 above)
Comment (jinevening) [reply compiler/quantization-value-test/CMakeLists.txt:136]: > Code itself looks good to me... but the name seems to be too general ``)

Our quantization scheme can be described as post-training affine quantization. But post-training-affine-quantization-value-test is too long, so how about using `paq-value-test` as the directory name? Each test names will be `paq_fake_quant_test`, `paq_record_minmax_test`, `paq_quantization_test`

 @seanshpark What do you think of this?

Comment (seanshpark) [reply compiler/quantization-value-test/CMakeLists.txt:136]: pronouncing is somewhat strange with `paq`... as word is `post-training-affine`, I could see `pta`.
- `pta_fake_quant_test`, `pta_record_minmax_test`, `pta_quantization_test`

Comment (jinevening) [reply compiler/quantization-value-test/CMakeLists.txt:136]: I thought `pta` before, but it is the abbreviation of "please try again".
(https://en.dict.naver.com/#/entry/enko/a0aeb300a9704137baa73fff43df1c47)

I think "please try again" quantization sounds that our tool does not work well..

Comment (seanshpark) [reply compiler/quantization-value-test/CMakeLists.txt:136]: Just `pa-` ?

Comment (jinevening) [reply compiler/quantization-value-test/CMakeLists.txt:136]: > Just pa- ?

Ok. I will go with `pa-quantization-value-test`. Cmake test names will be `pa_fake_quant_test`, `pa_record_minmax_test`, `pa_quantization_test`.

File: compiler/pota-quantization-value-test/CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+  DEPENDS record-minmax
+  DEPENDS circle2circle
+  DEPENDS circle-tensordump
+  COMMENT "Generate test configuration"
+)
+
+list(APPEND TEST_DEPS "${TEST_CONFIG}")
+
+# This enforces CMake to generate all the dependencies during "build" phase
+add_custom_target(quantization_value_test_deps ALL DEPENDS ${TEST_DEPS} pota-quantization-value-test_python_deps)
```
Comment (seanshpark) [inline compiler/pota-quantization-value-test/CMakeLists.txt:108]: maybe rename target `quantization_value_test_deps ` to `pota_quantization_value_test_deps ` ?

Comment (jinevening) [reply compiler/pota-quantization-value-test/CMakeLists.txt]: I've changed the target name. Thanks.

**Pull request merged**